subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/impl/DirectorySnapshotterTest.groovy

```diff
@@ -259,12 +259,45 @@ class DirectorySnapshotterTest extends Specification {
         then:
         relativePaths == ["included", "included/text.txt", "includedSymlink", "includedSymlink/included.txt", "includedSymlink/symlinkedFile.txt"]
         SnapshotVisitorUtil.getAbsolutePaths(snapshot) == relativePaths.collect { new File(rootDir, it).absolutePath }
-        completeSubsnapshots*.absolutePath == [
-            // use "new File()" since TestFile canonicalizes the paths.
-            new File(rootDir, "includedSymlink/included.txt"),
-            new File(rootDir, "includedSymlink/symlinkedFile.txt"),
-            new File(rootDir, "included")
-        ]*.absolutePath
+        completeSubsnapshots*.absolutePath == absolutePathsFor(rootDir,
+            "includedSymlink/included.txt",
+            "includedSymlink/symlinkedFile.txt",
+            "included")
+    }
+
+    @Requires(TestPrecondition.SYMLINKS)
+    def "partially snapshotted symlinked directories are handled correctly"() {
```

Comment:
Not sure what exactly to say here, but it would be good to give more context than "handled correctly" for later readers if possible.

Reply:
I improved the name.